packages/forms/src/UIForm/fields/Enumeration/EnumerationWidget.js

```diff
@@ -0,0 +1,1043 @@
+				onClick: this.onDeleteItems.bind(this),
+			},
+		];
+
+		let defaultDisplayMode = enumerationStates.DISPLAY_MODE_DEFAULT;
+		if (props.schema && props.schema.displayMode) {
+			defaultDisplayMode = props.schema.displayMode;
+		}
+
+		this.state = {
+			inputRef: this.setInputRef.bind(this),
+			displayMode: defaultDisplayMode,
+			searchCriteria: '',
+			required: (props.schema && props.schema.required) || false,
+			headerDefault: this.defaultHeaderActions,
+			headerSelected: this.selectedHeaderActions,
+			headerInput: this.addInputs,
+			items: (props.value || []).map(item => ({
+				id: item.id,
+				values: item.values,
+			})),
+			itemsProp: {
+				key: 'values',
+				getItemHeight: this.constructor.getItemHeight.bind(this),
+				onSubmitItem: this.onSubmitItem.bind(this),
+				onAbortItem: this.onAbortItem.bind(this),
+				onChangeItem: this.onChangeItem.bind(this),
+				onSelectItem: !disabledAction ? this.onSelectItem.bind(this) : () => {},
+				onLoadData: this.onLoadData.bind(this),
+				actionsDefault: this.defaultActions,
+				actionsEdit: this.itemEditActions,
+			},
+			onInputChange: this.onInputChange.bind(this),
+			onAddKeyDown: this.onAddKeyDown.bind(this),
+		};
+		this.onBlur = this.onBlur.bind(this);
+	}
+
+	componentWillReceiveProps(nextProps) {
+		if (nextProps.value) {
+			this.setState(prevState => ({ ...prevState, items: nextProps.value }));
+		}
+	}
+
+	onBlur(event) {
+		const { schema, onFinish } = this.props;
+		onFinish(event, { schema });
+	}
+
+	onChange(event, payload) {
+		const { schema, onFinish, onChange } = this.props;
+		onChange(event, payload);
+		onFinish(event, { schema });
+	}
+
+	onImportAppendClick() {
+		this.setState(
+			state => ({ ...state, importMode: enumerationStates.IMPORT_MODE_APPEND }),
+			this.simulateClickInputFile.bind(this),
+		);
+	}
+
+	onImportOverwriteClick() {
+		this.setState(
+			state => ({ ...state, importMode: enumerationStates.IMPORT_MODE_OVERWRITE }),
+			this.simulateClickInputFile.bind(this),
+		);
+	}
+
+	// default mode
+	onEnterEditModeItem(event, value) {
+		this.setState(prevState => {
+			let items = resetItems([...prevState.items]);
+			let item = items[value.index];
+			// if there is a search criteria, retrieve correct item from state in non-connected mode
+			if (prevState.searchCriteria && !this.isConnectedMode()) {
+				item = this.getItemInSearchMode(prevState.searchCriteria, value.index, items);
+			}
+			item.displayMode = enumerationStates.DISPLAY_MODE_EDIT;
+			// resetting errors
+			items[value.index].error = '';
+			// reset selection
+			items = items.map(currentItem => ({ ...currentItem, isSelected: false }));
+			// exit from selected mode to not display 0 values selected
+			let displayMode = prevState.displayMode;
+			if (displayMode === enumerationStates.DISPLAY_MODE_SELECTED) {
+				displayMode = enumerationStates.DISPLAY_MODE_DEFAULT;
+			}
+			const validation = this.constructor.updateItemValidateDisabled(item.values[0]);
```

Comment:
please review the use of all theses static methods.
Do you need them to be attached to the component. You can use directly Enumeration.myStatic